amr-wind/utilities/sampling/Sampling.cpp

```diff
@@ -65,25 +65,47 @@ void Sampling::initialize()
         m_samplers.emplace_back(std::move(obj));
     }
 
+    update_container();
+
+    if (m_out_fmt == "netcdf") prepare_netcdf_file();
+}
+
+void Sampling::update_container()
+{
+    BL_PROFILE("amr-wind::Sampling::update_container");
+
     // Initialize the particle container based on user inputs
     m_scontainer.reset(new SamplingContainer(m_sim.mesh()));
-    m_scontainer->setup_container(ncomp);
+    m_scontainer->setup_container(m_ncomp);
     m_scontainer->initialize_particles(m_samplers);
     // Redistribute particles to appropriate boxes/MPI ranks
     m_scontainer->Redistribute();
     m_scontainer->num_sampling_particles() = m_total_particles;
+}
 
-    if (m_out_fmt == "netcdf") prepare_netcdf_file();
+void Sampling::pre_advance_work()
+{
+    BL_PROFILE("amr-wind::Sampling::pre_advance_work");
+
+    SamplerBase::SampleLocType locs;
+    for (const auto& obj : m_samplers) {
+        obj->update_sampling_locations();
+    }
+
+    update_container();
 }
 
 void Sampling::post_advance_work()
 {
+
     BL_PROFILE("amr-wind::Sampling::post_advance_work");
     const auto& time = m_sim.time();
     const int tidx = time.time_index();
     // Skip processing if it is not an output timestep
     if (!(tidx % m_out_freq == 0)) return;
 
+    pre_advance_work();
+
```

Comment:
It is misleading to call `pre_advance_work` within `post_advance_work`. Please fix this. Looking at the logic, it appears that we don't need to add `pre_advance_work` but instead rename the function `update_sampling_locations` in `Sampling` class too.